### ISSUE TITLE
Normalize escaping in docstrings

### ISSUE THREAD

**Issue by jberdine** [ocaml-ppx/ocamlformat#482]
When parsing and formatting docstrings, escape sequences are sometimes expanded, causing the check to fail. For example on the following docstrings from `ocaml/ocamldoc/odoc_info.mli`
```
(** to extend \{foo syntax *)
(** The different forms of references in \@see tags. *)
```

**Comment by gpetiot** [comment]
You only have this error on branch #481 do you?

**Comment by jberdine** [comment]
I have only only tested the docstring parsing with #480 and #481, so you could be right.

**Comment by gpetiot** [comment]
Seems to be fixed by replacing
```
Format_.asprintf "%a@!"     ...
```
by
```
Format_.asprintf "%a%!"     ...
```
The fix will be done in #481.

**Comment by jberdine** [comment]
There are still some issues with escaping in docstrings.  One failure is on code/ocaml/ocamldoc/odoc_man.ml, where the ast changes as follows:
```
------ odoc_man.old.ast
++++++ odoc_man.new.ast
@|-559,33 +559,33 ============================================================
 |                                            expression (_none_[1,0+-1]..[1,0+-1]) ghost
 |                                              Pexp_ident "Buffer.contents" (_none_[1,0+-1]..[1,0+-1]) ghost
 |                                            [
 |                                              <arg>
 |                                              Nolabel
 |                                                expression (_none_[1,0+-1]..[1,0+-1]) ghost
 |                                                  Pexp_ident "b" (_none_[1,0+-1]..[1,0+-1]) ghost
 |                                            ]
 |                          ]
 |                      None
 |                class_field (_none_[1,0+-1]..[1,0+-1]) ghost
 |                    attribute "ocaml.doc"
 |                      [
 |                        structure_item (_none_[1,0+-1]..[1,0+-1]) ghost
 |                          Pstr_eval
 |                          expression (_none_[1,0+-1]..[1,0+-1]) ghost
-|                            Pexp_constant PConst_string("Printf groff string for the @before information.",None)
+|                            Pexp_constant PConst_string("Printf groff string for the\n@before information. ",None)
 |                      ]
 |                  Pcf_method Public
 |                    "str_man_of_before" (_none_[1,0+-1]..[1,0+-1]) ghost
 |                    Concrete Fresh
 |                    expression (_none_[1,0+-1]..[1,0+-1]) ghost
 |                      Pexp_poly
 |                      expression (_none_[1,0+-1]..[1,0+-1]) ghost
 |                        Pexp_function
 |                        [
 |                          <case>
 |                            pattern (_none_[1,0+-1]..[1,0+-1]) ghost
 |                              Ppat_construct "[]" (_none_[1,0+-1]..[1,0+-1]) ghost
 |                              None
 |                            expression (_none_[1,0+-1]..[1,0+-1]) ghost
 |                              Pexp_constant PConst_string("",None)
 |                          <case>
```

**Comment by gpetiot** [comment]
Fixed with #511
